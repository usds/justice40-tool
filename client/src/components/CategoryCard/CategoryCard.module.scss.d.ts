declare namespace IndicatorCategoryNamespace {
    export interface IIndicatorCategoryScss {
      categoryCard: string;
      idAsDisdvantaged: string;
    }
  }

declare const IndicatorCategoryScssModule: IndicatorCategoryNamespace.IIndicatorCategoryScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: IndicatorCategoryNamespace.IIndicatorCategoryScss;
  };

  export = IndicatorCategoryScssModule;
