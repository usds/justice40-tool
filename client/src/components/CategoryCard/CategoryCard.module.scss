@use '../../styles/design-system.scss' as *;

@mixin baseCard {
  @include u-padding-top(0);
  @include u-padding-right(6);
  @include u-padding-bottom(6);
  @include u-padding-left(6);
  @include u-margin-bottom(6);
  max-width: 34rem;
}

.categoryCard {
  @include baseCard;
  @include u-bg('blue-cool-5');
}

.idAsDisdvantaged {
  font-style: italic;
}