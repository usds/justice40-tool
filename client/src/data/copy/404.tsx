import React from 'react';
import {defineMessages} from 'react-intl';
import {FormattedMessage} from 'gatsby-plugin-intl';
import {linkFn} from './common';
import {PAGES_ENDPOINTS} from '../constants';

export const PAGE = defineMessages({
  TITLE: {
    id: 'pageNotFound.title.text',
    defaultMessage: 'Page not found',
    description: 'page not found title text',
  },
  HEADING: {
    id: 'pageNotFound.heading.text',
    defaultMessage: 'Page not found',
    description: 'page not found heading text',
  },
  GUIDANCE: {
    id: 'pageNotFound.Guidance.text',
    defaultMessage: 'Try creating a page in',
    description: 'page not found guidance text',
  },
});

export const ERROR_MSG =
<FormattedMessage
  id={'pageNotFound.apology.text'}
  defaultMessage={`
    Sorry, the page you are looking for is not found. <link1>Explore the map</link1> or learn 
    more <link2>about</link2> the tool.
  `}
  description={'main error message'}
  values={{
    link1: linkFn(PAGES_ENDPOINTS.EXPLORE, true, false),
    link2: linkFn(PAGES_ENDPOINTS.ABOUT, true, false),
  }}
/>;
