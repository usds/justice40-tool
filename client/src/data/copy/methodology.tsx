import React from 'react';
import {defineMessages} from 'react-intl';
import {FormattedMessage} from 'gatsby-plugin-intl';

export const PAGE = defineMessages({
  TILE: {
    id: 'methodology.page.title.text',
    defaultMessage: 'Data and Methodology',
    description: 'methodology page title text',
  },
  HEADING: {
    id: 'methodology.page.header.text',
    defaultMessage: 'Methodology',
    description: 'methodology page header text',
  },
  DESCRIPTION: {
    id: 'methodology.page.paragraph.first',
    defaultMessage: 'The methodology for identifying communities of focus is currently ' +
      'in a draft, pre-decisional form that may change over time as more datasets become available.',
    description: 'methodology page paragraph 1',
  },
});


// Download Package
export const DOWNLOAD_FILE_SIZE = '143MB';
export const DOWNLOAD_LAST_UPDATED = '09/20/21';
export const VERSION_NUMBER = '0.1';

export const DOWNLOAD_ZIP_URL = [
  process.env.GATSBY_DATA_ROOT_PATH,
  process.env.GATSBY_DATA_PIPELINE_SCORE_PATH,
  process.env.GATSBY_SCORE_DOWNLOAD_FILE_PATH,
].join('/');

export const DOWNLOAD_PACKAGE = defineMessages({
  TITLE: {
    id: 'downloadPacket.header.text',
    defaultMessage: `Draft communities list v${VERSION_NUMBER} (${DOWNLOAD_FILE_SIZE})`,
    description: 'download packet header text',
  },
  DESCRIPTION: {
    id: 'downloadPacket.info.text',
    defaultMessage: `The package includes draft v${VERSION_NUMBER} `+
    ` of the list of communities of focus (.csv and .xlsx) ` +
    ` and information about how to use the list (.pdf).`,
    description: 'download packet info text',
  },
  LAST_UPDATED: {
    id: 'downloadPacket.info.last.updated',
    defaultMessage: `Last updated: ${DOWNLOAD_LAST_UPDATED} `,
    description: 'download packet info last updated',
  },
  BUTTON_TEXT: {
    id: 'downloadPacket.button.text',
    defaultMessage: 'Download package',
    description: 'download packet button text',
  },
});


// Dataset section
export const DATASETS = defineMessages({
  HEADING: {
    id: 'datasetContainer.heading',
    defaultMessage: 'Datasets used in methodology',
    description: 'section heading of which datasets are used in cumulative score',
  },
  INFO: {
    id: 'datasetContainer.info',
    defaultMessage: 'The datasets come from a variety of sources and were selected' +
    ' based on relevance, availability, recency, and quality. The datasets seek to' +
    ' identify a range of human health, environmental, climate-related, and other' +
    ' cumulative impacts on communities.',
    description: 'description of the dataset section',
  },
  ADDITIONAL_HEADING: {
    id: 'datasetContainer.additional.heading',
    defaultMessage: 'Additional Indicators',
    description: 'additional indicators heading',
  },
  ADDITIONAL_INFO: {
    id: 'datasetContainer.additional.info',
    defaultMessage: 'These datasets provide additional information about each community.',
    description: 'additional indicator info',
  },
});

export const DATASET_CARD_LABELS = defineMessages({
  RESOLUTION: {
    id: 'datasetCard.dataResolution',
    defaultMessage: 'Data resolution: ',
    description: 'label associated with explaining the card',
  },
  SOURCE: {
    id: 'datasetCard.dataSource',
    defaultMessage: 'Data source: ',
    description: 'label associated with explaining the card',
  },
  DATE_RANGE: {
    id: 'datasetCard.dataDateRange',
    defaultMessage: 'Data date range: ',
    description: 'label associated with explaining the card',
  },
});

export const INDICATORS = [
  {
    indicator: 'Area Median Income',
    description: `Median income of the census block group calculated as a percent 
    of the metropolitan area’s or state's median income.`,
    dataResolution: `Census block group`,
    dataSourceLabel: `Census's American Community Survey`,
    dataSourceURL: `https://www.census.gov/programs-surveys/acs`,
    dataDateRange: `2015-2019`,
  },
  {
    indicator: 'Poverty',
    description: `Percent of a block group's population in households where the household income` +
    ` is at or below 100% of the federal poverty level.`,
    dataResolution: `Census block group`,
    dataSourceLabel: `Census's American Community Survey`,
    dataSourceURL: `https://www.census.gov/programs-surveys/acs`,
    dataDateRange: `2015-2019`,
  },
  {
    indicator: 'Education, less than high school education',
    description: `Percent of people ages 25 years or older in a block group whose 
    education level is less than a high school diploma.`,
    dataResolution: `Census block group`,
    dataSourceLabel: `Census's American Community Survey`,
    dataSourceURL: `https://www.census.gov/programs-surveys/acs`,
    dataDateRange: `2015-2019`,
  },
];

export const ADDITIONAL_INDICATORS = [
  {
    indicator: 'Diabetes',
    description: `People ages 18 years and older who report having ever been 
    told by a doctor, nurse, or other health professionals that they have 
    diabetes other than diabetes during pregnancy.`,
    dataResolution: `Census tract`,
    dataSourceLabel: `Centers for Disease Control and Prevention (CDC) PLACES`,
    dataSourceURL: `https://www.cdc.gov/places/index.html`,
    dataDateRange: `2016-2019`,
  },
  {
    indicator: 'Asthma',
    description: `Weighted number of respondents people who answer “yes” both 
    to both of the following questions: “Have you ever been told by a doctor, 
    nurse, or other health professional that you have asthma?” and the question 
    “Do you still have asthma?”`,
    dataResolution: `Census tract`,
    dataSourceLabel: `Centers for Disease Control and Prevention (CDC) PLACES`,
    dataSourceURL: `https://www.cdc.gov/places/index.html`,
    dataDateRange: `2016-2019`,
  },
  {
    indicator: 'Heart disease',
    description: `People ages 18 years and older who report ever having been told 
    by a doctor, nurse, or other health professionals that they had angina or 
    coronary heart disease.`,
    dataResolution: `Census tract`,
    dataSourceLabel: `Centers for Disease Control and Prevention (CDC) PLACES`,
    dataSourceURL: `https://www.cdc.gov/places/index.html`,
    dataDateRange: `2016-2019`,
  },
  {
    indicator: 'Life expectancy',
    description: `Estimated years of life expectancy.`,
    dataResolution: `Census tract`,
    dataSourceLabel: `Centers for Disease Control and Prevention (CDC) 
    US Small-area Life Expectancy Estimates Project`,
    dataSourceURL: `https://www.cdc.gov/nchs/nvss/usaleep/usaleep.html#data`,
    dataDateRange: `2010-2015`,
  },
  {
    indicator: 'Traffic proximity and volume',
    description: `Count of vehicles (average annual daily traffic) at major roads 
    within 500 meters, divided by distance in meters (not km).`,
    dataResolution: `Census block group`,
    dataSourceLabel: `Department of Transportation (DOT) traffic data as compiled by EPA's EJSCREEN`,
    dataSourceURL: `https://www.epa.gov/ejscreen/technical-documentation-ejscreen`,
    dataDateRange: `2017`,
  },
  {
    indicator: 'FEMA Risk Index Expected Annual Loss Score',
    description: `Average economic loss in dollars resulting from natural 
    hazards each year. It is calculated for each hazard type and quantifies 
    loss for relevant consequence types: buildings, people, and agriculture.`,
    dataResolution: `Census tract`,
    dataSourceLabel: `Federal Emergency Management Agency (FEMA)`,
    dataSourceURL: `https://hazards.fema.gov/nri/expected-annual-loss`,
    dataDateRange: `2014-2017`,
  },
  {
    indicator: 'Energy burden',
    description: `Average annual energy cost ($) divided by household income.`,
    dataResolution: `Census tract`,
    dataSourceLabel: `Department of Energy (DOE) LEAD Score`,
    dataSourceURL: `https://www.energy.gov/eere/slsc/low-income-energy-affordability-data-lead-tool`,
    dataDateRange: `2018`,
  },
  {
    indicator: 'Housing cost burden',
    description: `Households that are low income and spend more than 30% of their 
    income to housing costs.`,
    dataResolution: `Census tract`,
    dataSourceLabel: `Department of Housing & Urban Development’s 
    (HUD) Comprehensive Housing Affordability Strategy dataset`,
    dataSourceURL: `https://www.huduser.gov/portal/datasets/cp.html`,
    dataDateRange: `2013-2017`,
  },
  {
    indicator: 'Wastewater discharge',
    description: `RSEI modeled Toxic Concentrations at stream segments within 500 
    meters, divided by distance in kilometers (km).`,
    dataResolution: `Census block group`,
    dataSourceLabel: `Environmental Protection Agency (EPA) Risk-Screening 
    Environmental Indicators (RSEI) Model as compiled by EPA's EJSCREEN`,
    dataSourceURL: `https://www.epa.gov/ejscreen/technical-documentation-ejscreen`,
    dataDateRange: `2020`,
  },
  {
    indicator: 'Lead paint',
    description: `Percent of housing units built pre-1960, used as an 
    indicator of potential lead paint exposure in homes.`,
    dataResolution: `Census block group`,
    dataSourceLabel: `Census's American Community Survey`,
    dataSourceURL: `https://www.census.gov/programs-surveys/acs`,
    dataDateRange: `2015-2019`,
  },
  {
    indicator: 'Diesel particulate matter',
    description: `Mixture of particles that is part of diesel exhaust in the air.`,
    dataResolution: `Census block group`,
    dataSourceLabel: `Environmental Protection Agency (EPA) National Air Toxics Assessment (NATA)
    as compiled by EPA's EJSCREEN`,
    dataSourceURL: `https://www.epa.gov/ejscreen/technical-documentation-ejscreen`,
    dataDateRange: `5-year estimates, 2015-2019`,
  },
  {
    indicator: 'PM2.5',
    description: `Fine inhalable particles, with diameters that are generally 
    2.5 micrometers and smaller.`,
    dataResolution: `Census block group`,
    dataSourceLabel: `Environmental Protection Agency (EPA) Office of Air 
    and Radiation (OAR) fusion of model and monitor data as compiled by EPA's EJSCREEN`,
    dataSourceURL: `https://www.epa.gov/ejscreen/technical-documentation-ejscreen`,
    dataDateRange: `2017`,
  },

];


// Methodology Steps:
export const METHODOLOGY_STEPS = defineMessages({
  HEADING: {
    id: 'methodology.steps.heading',
    defaultMessage: `Methodology`,
    description: 'heading of methodology section',
  },
  DESCRIPTION_1: {
    id: 'methodology.steps.description.1',
    defaultMessage: 'The methodology for identifying communities of focus is'+
     ' calculated at the census block group level. Census block geographical boundaries'+
     ' are determined by the U.S. Census Bureau once every ten years. This tool utilizes'+
     ' the census block boundaries from 2010.',
    description: 'first description text ',
  },
  DESCRIPTION_2: {
    id: 'methodology.steps.description.2',
    defaultMessage: 'The following describes the process for identifying communities of focus.',
    description: 'second description text',
  },
  STEP_1_HEADING: {
    id: 'methodology.step.1.heading',
    defaultMessage: `Gather datasets`,
    description: 'first step heading',
  },
  STEP_1_INFO: {
    id: 'methodology.step.1.info',
    defaultMessage: `The methodology includes the following inputs that are equally weighted.`,
    description: 'first step info',
  },
  STEP_1_A_HEADING: {
    id: 'methodology.step.1.a.heading',
    defaultMessage: `Percent of Area Median Income`,
    description: 'step 1 a heading',
  },
  STEP_1_A_INFO_1: {
    id: 'methodology.step.1.a.info.1',
    defaultMessage: 'If a census block group is in a metropolitan area, this value is the'+
    ' median income of the census block group calculated as a percent of'+
    ' the metropolitan area’s median income.',
    description: 'step 1 a info 1',
  },
  STEP_1_A_INFO_2: {
    id: 'methodology.step.1.a.info.2',
    defaultMessage: 'If a census block group is not in a metropolitan area, this value is the'+
    ' median income of the census block group calculated as a percent of the state’s median'+
    ' income.',
    description: 'step 1 a info 2',
  },
  STEP_1_B_HEADING: {
    id: 'methodology.step.1.b.heading',
    defaultMessage: `Percent of households below or at 100% of the federal poverty line`,
    description: 'step 1 b heading',
  },
  STEP_1_C_HEADING: {
    id: 'methodology.step.1.c.heading',
    defaultMessage: `The high school degree achievement rate for adults 25 years and older`,
    description: 'step 1 a heading',
  },
  STEP_1_C_INFO: {
    id: 'methodology.step.1.c.info',
    defaultMessage: 'The percent of individuals who are 25 or older who have received a high school degree.',
    description: 'step 1 c info',
  },
  STEP_2_HEADING: {
    id: 'methodology.step.2.heading',
    defaultMessage: `Determine communites of focus`,
    description: 'second step heading',
  },
  STEP_2_INFO: {
    id: 'methodology.step.2.info',
    defaultMessage: `Under the existing formula, a census block group will be considered a community of focus if:`,
    description: 'second step info',
  },
});

const FED_POVERTY_LINE_URL = 'https://www.census.gov/topics/income-poverty/poverty/guidance/poverty-measures.html';

// Copy that has links or other HTML tags in them
export const COMPLEX_METH_STEPS = {
  STEP_2_B_INFO: <FormattedMessage
    id={'methodology.steps.2.b.info'}
    description={'Download the draft list of communities of focus and datasets used.'}
    defaultMessage={`This is the percent of households in a state with a household income 
    below or at 100% of the {federalPovertyLine}. This federal poverty line is calculated 
    based on the composition of each household (e.g., based on household size), but it does not vary geographically.`}
    values={{
      federalPovertyLine:
      <a href={FED_POVERTY_LINE_URL} target="_blank" rel="noreferrer">
        federal poverty line
      </a>,
    }}
  />,
  FORMULA: <FormattedMessage
    id={'methodology.steps.2.formula'}
    description={'Formala used to calculate communities of focus'}
    defaultMessage={`{medianIncome} {or} {livingAtPovery} {and} {education}`}
    values={{
      medianIncome:
        <p>
         (The median income is less than 80% of the area median income
        </p>,
      or:
        <p className={'flush'}>
          OR
        </p>,
      livingAtPovery:
        <p className={'flush'}>
          households living in poverty (at or below 100% of the federal poverty level) is greater than 20%)
        </p>,
      and:
        <p className={'flush'}>
          AND
        </p>,
      education:
        <p className={'flush'}>
          The high school degree achievement rate for adults 25 years and older is greater than 95%
        </p>,
    }}
  />,
};
