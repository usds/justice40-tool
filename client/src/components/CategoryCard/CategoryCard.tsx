import React from 'react';

import * as METHODOLOGY_COPY from '../../data/copy/methodology';
import * as styles from './CategoryCard.module.scss';
interface ICategoryInterface {
    categoryInfo: {
        TITLE: JSX.Element,
        IF: JSX.Element,
        AND: JSX.Element
    }
}
const CategoryCard = ({categoryInfo}: ICategoryInterface) => {
  return (
    <div className={styles.categoryCard}>
      <h3>
        {categoryInfo.TITLE}
      </h3>
      <p className={styles.idAsDisdvantaged}>
        {METHODOLOGY_COPY.CATEGORY.ID_AS_DISADV_TEXT}
      </p>
      <p>
        {categoryInfo.IF}
      </p>
      <p>
        {categoryInfo.AND}
      </p>
    </div>
  );
};

export default CategoryCard;
