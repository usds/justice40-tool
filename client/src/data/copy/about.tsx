/* eslint-disable max-len */
import React from 'react';
import {defineMessages} from 'react-intl';
import {FormattedMessage} from 'gatsby-plugin-intl';

import {italicFn, linkFn} from './common';
import {VERSION_NUMBER} from './methodology';
import {PAGES_ENDPOINTS} from '../constants';

export const EXEC_ORDER_LINK = 'https://www.federalregister.gov/documents/2021/02/01/2021-02177/tackling-the-climate-crisis-at-home-and-abroad#:~:text=Sec.%20223.%20Justice40,40-percent%20goal.';
export const FAQS_LINK = 'https://www.whitehouse.gov/wp-content/uploads/2022/02/CEQ-CEJST-QandA.pdf';
export const FED_RECOGNIZED_INDIAN_ENTITIES = `https://www.federalregister.gov/documents/2022/01/28/2022-01789/indian-entities-recognized-by-and-eligible-to-receive-services-from-the-united-states-bureau-of`;
export const EJSCREEN = 'https://www.epa.gov/ejscreen/how-does-epa-use-ejscreen';

export const CEJST_INSTRUCT = `https://static-data-screeningtool.geoplatform.gov/data-versions/1.0/data/score/downloadable/CEQ-CEJST-Instructions.pdf`;
export const CEJST_MEMO = `https://www.whitehouse.gov/wp-content/uploads/2023/01/M-23-09_Signed_CEQ_CPO.pdf`;
export const PAGE = defineMessages({
  TITLE: {
    id: 'about.page.title.text',
    defaultMessage: 'About',
    description: 'Navigate to the About page. This is the about page title text',
  },
});

export const CONTENT = {
  PARA1:
  <FormattedMessage
    id={'about.page.paragraph.1'}
    defaultMessage={`
      In January of 2021, President Biden issued <link1>Executive Order 14008</link1>. The order directed the Council on Environmental Quality (CEQ) to develop a new tool. This tool is called the Climate and Economic Justice Screening Tool. The tool has an interactive map and uses datasets that are indicators of burdens in eight categories: climate change, energy, health, housing, legacy pollution, transportation, water and wastewater, and workforce development. The tool uses this information to identify communities that are experiencing these burdens. These are the communities that are disadvantaged because they are overburdened and underserved.

    `}
    description={'Navigate to the About page. This is the paragraph 1'}
    values={{
      link1: linkFn(EXEC_ORDER_LINK, false, true),
      italictag: italicFn,
    }}
  />,
  PARA2:
  <FormattedMessage
    id={'about.page.paragraph.2'}
    defaultMessage={`
      Federal agencies will use the tool to help identify disadvantaged communities that will benefit from programs included in the <link1>Justice40 Initiative</link1>. The Justice40 Initiative seeks to deliver 40% of the overall benefits of investments in climate, clean energy, and related areas to disadvantaged communities.
    `}
    description={'Navigate to the About page. This is the paragraph 2'}
    values={{
      link1: linkFn(EXEC_ORDER_LINK, false, true),
      italictag: italicFn,
    }}
  />,
  PARA3:
  <FormattedMessage
    id={'about.page.paragraph.3'}
    defaultMessage={`
      Federal agencies should also use the following:
    `}
    description={'Navigate to the About page. This is the paragraph 3'}
  />,
  LI1:
  <FormattedMessage
    id={'about.page.list.item.1'}
    defaultMessage={`
      <link1>Memorandum</link1> on Using the CEJST for the Justice40 Initiative
    `}
    description={'Navigate to the About page. This is the list item 1'}
    values={{
      link1: linkFn(CEJST_MEMO, false, true),
    }}
  />,
  LI2:
  <FormattedMessage
    id={'about.page.list.item.2'}
    defaultMessage={`
     <link1>Instructions</link1> to Federal agencies on using the CEJST
    `}
    description={'Navigate to the About page. This is the list item 2'}
    values={{
      link1: linkFn(CEJST_INSTRUCT, false, true),
    }}
  />,
  PARA4:
  <FormattedMessage
    id={'about.page.paragraph.4'}
    defaultMessage={`
      CEQ will update the tool each year based on public feedback, research, and the availability of new data. The current version of the tool is version {version}. <link1>Sign-up</link1> for updates from CEQ
    `}
    description={'Navigate to the About page. This is the paragraph 4'}
    values={{
      version: VERSION_NUMBER,
      link1: linkFn('https://lp.constantcontactpages.com/su/Vm8pCFj/spring', false, true),
    }}
  />,
  PARA5:
  <FormattedMessage
    id={'about.page.paragraph.5'}
    defaultMessage={`
      A Spanish version of the site will be available in the near future.
    `}
    description={'Navigate to the About page. This is the paragraph 5'}
  />,
  USE_DATA_PARA:
  <FormattedMessage
    id={'about.page.use.data.paragraph'}
    defaultMessage={`
      The tool's data is available for <link1>download</link1>. This data can be used to filter by state or county. 
    `}
    description={'Navigate to the About page. This is the paragraph 4'}
    values={{
      link1: linkFn(PAGES_ENDPOINTS.DOWNLOADS, true, false),
    }}
  />,
  HOW_TO_USE_PARA1:
  <FormattedMessage
    id={'about.page.how.to.use.tool.para1'}
    defaultMessage={`
      The tool shows information about the burdens that communities experience. It uses datasets to identify indicators of burdens. The tool shows these burdens in census tracts. Census tracts are small units of geography. Census tract boundaries for <link1>statistical areas</link1> are determined by the U.S. Census Bureau once every ten years. The tool utilizes the census tract boundaries from 2010. This was chosen because many of the data sources in the tool currently use the 2010 census boundaries.  The tool also shows land within the boundaries of Federally Recognized Tribes and point locations for Alaska Native Villages.
    `}
    description={'Navigate to the About page. This is the paragraph 4'}
    values={{
      link1: linkFn('https://www.census.gov/programs-surveys/acs/geography-acs/geography-boundaries-by-year.html', false, true),
    }}
  />,
  HOW_TO_USE_PARA3:
  <FormattedMessage
    id={'about.page.how.to.use.para3'}
    defaultMessage={`
      A community is considered to be disadvantaged if they are located within a census tract that meets the tool’s <link1>methodology</link1> or are on land within the boundaries of Federally Recognized Tribes.
    `}
    description={'Navigate to the About page. This is the paragraph 4'}
    values={{
      link1: linkFn(PAGES_ENDPOINTS.METHODOLOGY, true, false),
    }}
  />,
};

export const GITHUB_LINK = 'https://github.com/usds/justice40-tool';

export const HOW_TO_USE_TOOL = defineMessages({
  TITLE: {
    id: 'about.page.how.to.use.tool.title',
    defaultMessage: 'How to use the tool',
    description: 'Navigate to the About page. This is the sub heading of How to use the tool',
  },
  PARA2: {
    id: 'about.page.how.to.use.tool.para2',
    defaultMessage: 'The tool ranks most of the burdens using percentiles. Percentiles show how much burden each tract experiences compared to other tracts. Certain burdens use percentages or a simple yes/no.',
    description: 'Navigate to the About page. This is the sub heading of How to use the tool paragraph1',
  },
  USE_MAP_HEADING: {
    id: 'about.page.use.map.heading',
    defaultMessage: 'Using the map',
    description: 'Navigate to the About page. This is the sub heading of Using the map',
  },
  USE_MAP_PARA: {
    id: 'about.page.use.map.para',
    defaultMessage: `
      Zoom in and select any census tract to see if it is considered disadvantaged.
    `,
    description: 'Navigate to the About page. This is the paragraph of Using the map',
  },
  USE_DATA_HEADING: {
    id: 'about.page.use.data.heading',
    defaultMessage: 'Using the data',
    description: 'Navigate to the About page. This is the sub heading of Using the data',
  },
});

export const GET_INVOLVED = defineMessages({
  TITLE: {
    id: 'about.page.getInvolved.title',
    defaultMessage: 'Get involved',
    description: 'Navigate to the About page. This is the sub heading of page',
  },
  SEND_FEEDBACK_HEADING: {
    id: 'about.page.send.feedback.heading',
    defaultMessage: 'Send feedback',
    description: 'Navigate to the About page. This is the sending feedback heading',
  },
  SEND_FEEDBACK_INFO: {
    id: 'about.page.send.feedback.info',
    defaultMessage: `
      Have ideas for the tool? Contact the Council on Environmental Quality’s (CEQ).
    `,
    description: 'Navigate to the About page. This is the sending feedback information',
  },
  JOIN_OSC_HEADING: {
    id: 'about.page.join.opensource.heading',
    defaultMessage: 'Join the open source community',
    description: 'Navigate to the About page. This is the join the community heading',
  },
  JOIN_OSC_INFO: {
    id: 'about.page.join.open.source.info',
    defaultMessage: `
      The tool’s code is open source, which means it is available for the public to view and contribute to it.
    `,
    description: 'info on Navigate to the About page. This is the joining open source community',
  },
  JOIN_OSC_LINK_TEXT: {
    id: 'about.page.join.open.source.link',
    defaultMessage: 'Check it out on GitHub',
    description: 'Navigate to the About page. This is the link to github repository',
  },
});

export const GET_INVOLVED_COMMENTS = {
  EMAIL:
  <FormattedMessage
    id={'about.page.send.feedback.email.link'}
    description={'about page sub header text'}
    defaultMessage={`
      Contact
    `}
  />,
};

export const HOW_YOU_CAN_HELP_LIST_ITEMS = {
  HEADING: <FormattedMessage
    id={'about.page.how.you.can.help.heading'}
    defaultMessage={`How you can help improve the map`}
    description={`Navigate to the about page. You will see How you can help`}
  />,
  LIST_ITEM_1: <FormattedMessage
    id={'about.page.how.you.can.help.list.item.1'}
    defaultMessage={`Provide <link1>general feedback</link1> on the CEJST website`}
    description={`Navigate to the about page. You will see How you can help list item 1`}
    values={{
      link1: linkFn('https://www.surveymonkey.com/r/P3LWTSB', false, true),
    }}
  />,
  LIST_ITEM_2: <FormattedMessage
    id={'about.page.how.you.can.help.list.item.2'}
    defaultMessage={`Suggest new <link1>data sources</link1>.`}
    description={`Navigate to the about page. You will see How you can help list item 2`}
    values={{
      link1: linkFn('https://www.surveymonkey.com/r/6G9TQJ8', false, true),
    }}
  />,
  LIST_ITEM_3: <FormattedMessage
    id={'about.page.how.you.can.help.list.item.3'}
    defaultMessage={`Have feedback about a specific census tract? You can either click here or click the “Send Feedback” button on the side-panel of a census tract on the map.`}
    description={`Navigate to the about page. You will see How you can help list item 3`}
  />,
  LIST_ITEM_4: <FormattedMessage
    id={'about.page.how.you.can.help.list.item.4'}
    defaultMessage={`Any other questions? The best way to contact the Council on Environmental Quality (CEQ) is by filling out this <link2>form</link2> . Otherwise, email: <link1>Screeningtool-Support@omb.eop.gov</link1>`}
    description={`Navigate to the about page. You will see How you can help list item 3`}
    values={{
      link1: linkFn(PAGES_ENDPOINTS.METHODOLOGY, true, false),
      link2: linkFn('https://www.surveymonkey.com/r/5LZ7MNB', false, true),
    }}
  />,
  PARA1: <FormattedMessage
    id={'about.page.how.you.can.help.para.1'}
    defaultMessage={`
      The Council on Environmental Quality plans to issue a Request for Information in 2023. This will give the public time to use the tool before providing comments.    
    `}
    description={`Navigate to the about page. You will see How you can help list item 3`}
  />,

};
