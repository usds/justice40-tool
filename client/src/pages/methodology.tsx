import * as React from 'react';
import {Grid} from '@trussworks/react-uswds';
import {useIntl} from 'gatsby-plugin-intl';
import {useWindowSize} from 'react-use';

import Categories from '../components/Categories';
import DatasetContainer from '../components/DatasetContainer';
import J40MainGridContainer from '../components/J40MainGridContainer';
import Layout from '../components/layout';
import SubPageNav from '../components/SubPageNav';
import PublicEngageButton from '../components/PublicEngageButton';

import {USWDS_BREAKPOINTS} from '../data/constants';
import * as METHODOLOGY_COPY from '../data/copy/methodology';
import {PAGES_ENDPOINTS} from '../data/constants';

interface MethodPageProps {
  location: Location;
}

const IndexPage = ({location}: MethodPageProps) => {
  const intl = useIntl();
  const {width} = useWindowSize();

  return (
    <Layout location={location} title={intl.formatMessage(METHODOLOGY_COPY.PAGE.TILE)}>

      <J40MainGridContainer>

        <section className={'page-heading'}>
          <h1>{intl.formatMessage(METHODOLOGY_COPY.PAGE.HEADING)}</h1>
          <PublicEngageButton />
        </section>

        <Grid row gap className={'j40-mt3'}>

          {/* First column */}
          <Grid col={12} tablet={{col: 8}}>
            <section>
              <p>
                {intl.formatMessage(METHODOLOGY_COPY.PAGE.PARA1)}
              </p>

              <div>
                <ul>
                  <li>
                    {intl.formatMessage(METHODOLOGY_COPY.PAGE.PARA1_BULLET1)}
                  </li>
                  <li className={'j40-mt3'}>
                    {intl.formatMessage(METHODOLOGY_COPY.PAGE.PARA1_BULLET2)}
                  </li>
                </ul>
              </div>
            </section>
          </Grid>

          {/* Second column */}
          <Grid col={12} tablet={{col: 1}}>
            {/* Spacer column */}
          </Grid>

          {/* Third column */}
          {width > USWDS_BREAKPOINTS.DESKTOP ?
          <Grid col={12} tablet={{col: 3}}>
            <SubPageNav
              endPoints={[
                PAGES_ENDPOINTS.METHODOLOGY,
                PAGES_ENDPOINTS.DOWNLOADS,
                PAGES_ENDPOINTS.PREVIOUS_VERSIONS,
              ]}
            />
          </Grid> : ''}
        </Grid>
      </J40MainGridContainer>

      <Categories />
      <DatasetContainer/>

    </Layout>
  );
};

export default IndexPage;
