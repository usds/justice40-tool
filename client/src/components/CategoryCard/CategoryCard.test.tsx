import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import CategoryCard from './CategoryCard';

import * as METHODOLOGY_COPY from '../../data/copy/methodology';

describe('rendering of the CategoryCard', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <CategoryCard categoryInfo={METHODOLOGY_COPY.CATEGORIES.CLIMATE_CHANGE}/>
      </LocalizedComponent>,
  );

  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
