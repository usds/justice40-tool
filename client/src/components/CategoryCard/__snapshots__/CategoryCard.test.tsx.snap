// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the CategoryCard checks if component renders 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Climate change
    </h3>
    <p>
      
      Communities are 
      <strong>
        identified as disadvantaged
      </strong>
       if they are in census tracts that:
    
    </p>
    <p>
      <strong>
        ARE
      </strong>
       at or above the 90th percentile for 
      <a
        class="usa-link"
        href="#exp-agr-loss-rate"
      >
        expected agriculture loss rate
      </a>
       OR 
      <a
        class="usa-link"
        href="#exp-bld-loss-rate"
      >
        expected building loss rate
      </a>
       OR 
      <a
        class="usa-link"
        href="#exp-pop-loss-rate"
      >
        expected population loss rate
      </a>
        OR 
      <a
        class="usa-link"
        href="#flood-risk"
      >
        projected flood risk
      </a>
       OR 
      <a
        class="usa-link"
        href="#wildfire-risk"
      >
        projected wildfire risk
      </a>
    </p>
    <p>
      
      
      <strong>
        AND
      </strong>
       are at or above the 65th percentile for 
      <a
        class="usa-link"
        href="#low-income"
      >
        low income
      </a>
      
    
    </p>
  </div>
</DocumentFragment>
`;
